app/reflexes/products_reflex.rb

```diff
@@ -78,13 +96,38 @@ def render_products
                              category_options: categories, category_id: @category_id })
     ).broadcast
 
+    render_product_delete_modals
+    render_variant_delete_modals
```

Comment:
Yes, I agree but I got the same question when I tried. I explored some but couldn't get a satisfactory answer. Let me dig into it.

Reply:
I just came across an example of sending a parameter. 
https://github.com/openfoodfoundation/openfoodnetwork/pull/11937/commits/4eb273b06ed941154b040962215620d0e23a7356#diff-690bba5275ec68bcde618132878b6504d984428b737915bf3f5e21d9ae841205R8

So I think it could be a `data-product_id="123"` attribute on the element, and the reflex can access it with `element.dataset.product_id`.

Reply:
Thank you, David 👍🏻

Reply:
I worked on the suggested changes and here are my findings on it:

- As per the current changes, the link also calls the stimulus action `click->modal-link#open`:

https://github.com/openfoodfoundation/openfoodnetwork/blob/d322f7240bb0ab565488190658d03959517c0f13/app/views/admin/products_v3/_table.html.haml#L73-L75

- This action call is needed to open the modal.
- If we add the reflex action here which renders the modal first, we may not get the expected result
- Stimulus regular action triggers on the client side and the Reflex action gets triggered on the server side
- So client side of the action will always be faster than the server side action and we will not have the desired modal rendered before the modal open call gets triggered
- Hence we won't be getting the modal in the `modal-link` controller to be opened
- I came up with the following changes:
```erb
%p{ "data-controller": "modal-link", "data-action": "click->modal-link#open",
    "data-modal-link-target-value": "delete_product_#{product.id}", "class": "delete",
    "data-object-id": product.id, "data-object-type": 'product', "data-reflex": 'mousedown->products#render_delete_modal' }
```
- I used the `mousedown` event but still it's slower😅. If I hold the mouse button for a fraction of second, and release it only then it's rendering and opening the modal.

- I think this needs architectural changes in `modal-link` controller where before opening the modal, we are stimulating a reflex to render it. If that's the case, I believe we should handle it in a separate issue.
- Please share your thoughts on it. Thanks.

Reply:
Thanks for exploring on this. I like the final solution you came up with instead 👍